GillianCore/utils/generators.ml

```diff
@@ -28,8 +28,11 @@ let fresh_lvar, reset_lvar = fresh_sth lvar_prefix
 (** Logical variable counter *)
 let fresh_lvar_bi, reset_lvar_bi = fresh_sth lvar_prefix_bi
 
+let fresh_svar, reset_svar = fresh_sth "#gen__"
```

Comment:
How does this prefix interact with, say, `#var` or just `#` later on?

Reply:
I tried to choose a prefix that wasn't used anywhere else, but that could be broken by a future instantiation
I believe we should intern all variable names, and that would make it easier to ensure freshness when generating them
But even if it can be broken in the future, this is still more resilient than what was there before

Reply:
Got it now, #var just doesn't exist